internal/webhooks/resources.go

```diff
@@ -0,0 +1,116 @@
+package webhooks
+
+import (
+	"context"
+	"strings"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/semconv"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func serviceName(pod corev1.Pod, resources map[string]string) string {
+	if name := resources[string(semconv.K8SDeploymentNameKey)]; name != "" {
```

Comment:
What about looking at `app` (probably along with `app.kubernetes.io/name`) label as well? I would even prioritize it over deployment/statefulset/... names. This is what istio uses as well

Reply:
Added

Reply:
I think we use `app` as well, `app.kubernetes.io/name` is recommended, but `app` is still widely used. WDYT?

Reply:
done